spring-jdbc/src/main/java/org/springframework/jdbc/core/simple/AbstractJdbcUpdate.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.jdbc.core.simple;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import javax.sql.DataSource;
+
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+
+/**
+ * Abstract class to provide base functionality for easy updates based on
+ * configuration options and database metadata. This class provides the base SPI
+ * for {@link SimpleJdbcUpdate}.
+ *
+ * @author Thomas Risberg
+ * @author Florent Paillard
+ * @author Sanghyuk Jung
+ * @since 4.3
+ */
+public abstract class AbstractJdbcUpdate extends AbstractSimpleSql {
+
+	/** The generated string used for update statement */
+	private String updateString;
+
+	/** The SQL type information for the declared columns */
+	private int[] columnTypes;
+
+	/** List of columns effectively used in 'set' clause */
+	private final List<String> reconciledUpdatingColumns = new ArrayList<String>();
+
+	/** The names of the columns to be used in 'where' clause */
+	private final List<String> restrictingColumns = new ArrayList<String>();
+
+
+	/**
+	 * Constructor for sublasses to delegate to for setting the DataSource.
+	 */
+	protected AbstractJdbcUpdate(DataSource dataSource) {
+		super(dataSource);
+	}
+
+	/**
+	 * Constructor for sublasses to delegate to for setting the JdbcTemplate.
+	 */
+	protected AbstractJdbcUpdate(JdbcTemplate jdbcTemplate) {
+		super(jdbcTemplate);
+	}
+
+
+	// -------------------------------------------------------------------------
+	// Methods dealing with configuration properties
```

Comment:
@tomsquest
I fixed typo ( configuaration -> configuration ), and rebased on master again.